include/image.mk

```diff
@@ -430,6 +433,9 @@ endef
 
 define Device/Check
   $(Device/Check/Common)
+  ifeq ($$(DEVICE_TITLE),)
+	  DEVICE_TITLE := $$(DEVICE_VENDOR) $$(DEVICE_MODEL)$$(if $$(value DEVICE_REVISION), $$(DEVICE_REVISION),)
+  endif
```

Comment:
If you use `=` intead of `:=` you should be able to put it into `Device/InitProfile`. `=` causes referenced variables to be evaluated at use time, not at definition time. See https://www.gnu.org/software/make/manual/html_node/Flavors.html#Flavors for more info.

Also you don't need to have a comma for the else branch if it's empty, having `$(if foo, bar)` is perfectly fine and valid.

Reply:
Updated, please check

Reply:
Is the `$$(value DEVICE_REVISION)` needed here? Not sure if just a `$$(if $$(DEVICE_REVISION),...)` would be enough.

Also might be reasonable to add those new variables to `DEVICE_VARS`.

Reply:
Thanks, removed it and still works.
Also added them to DEFAULT_DEVICE_VARS